### ISSUE TITLE
alpakka-jms: boxed properties causing exception

### ISSUE THREAD

**Issue by IndiscriminateCoding** [akka/alpakka#2365]
### Versions used 

"com.lightbend.akka" %% "akka-stream-alpakka-jms" % "2.0.1"

Akka version: transient dependency

### Expected Behavior

Setting JMS properties of type other than a `String` works reliably

### Actual Behavior

Settings JMS properties of primitive types causes `akka.stream.alpakka.jms.UnsupportedMessagePropertyType` exception

### The problem

JMS allow primitive types to be used as a property.

However, alpakka-jms do following:
1) explicitly checks for unboxed primitives (or `String`) to be used;
2) stores JMS properties in a `Map[String, Any]`

Combination of [1] and [2] makes it impossible to **reliably** use primitive properties - if scala compiler decides to box a value (causing it to be of type `java.lang.Float`, for example), it will throw an exception.

Problematic code (that should have cases for `java.lang._` types) is [here](https://github.com/akka/alpakka/blob/master/jms/src/main/scala/akka/stream/alpakka/jms/impl/JmsMessageProducer.scala#L76), I suppose.

**Comment by ennru** [comment]
> Problematic code (that should have cases for java.lang._ types) is here, I suppose.

Yes, that sounds correct to me. Would you be in a position to suggest the neccessary changes in a pull request?

**Comment by IndiscriminateCoding** [comment]
@ennru I tried to; but building alpakka gives me compilation errors which I don't know how to solve

**Comment by ennru** [comment]
Sorry to hear that. Did you build it with `sbt Test/compile` before trying to use an IDE?
Can you post some of the errors so we can try to figure it out?

**Comment by IndiscriminateCoding** [comment]
@ennru the error I encounter on `sbt Test/compile` is:
```
[error] /home/me/Projects/alpakka/jms/src/main/scala/akka/stream/alpakka/jms/AcknowledgeMode.scala:7:8: object jms is not a member of package javax
[error] import javax.jms
[error]        ^
```

**Comment by ennru** [comment]
Hm, I don't know what could cause this problem. `"javax.jms" % "jms" % "1.1"` is marked as `provided` in `Dependencies.scala`. You could try to remove that marker.

**Comment by ennru** [comment]
I'm looking into this. It is just that `Float` is missing in the pattern matches, Scala does the right thing about boxing into the Java types.

**Comment by ennru** [comment]
Fixed with #2408
